src/pages/Book/index.js

```diff
@@ -4,35 +4,81 @@ import Cover from "../../components/Cover";
 import Title from "../../components/Title";
 import Author from "../../components/Author";
 import Description from "../../components/Description";
-import { Container, CoverDiv } from "./book.styles";
+import Loading from "../../components/Loading";
+import { Container, CoverDiv, Favorite } from "./book.styles";
+import { FaStar, FaRegStar } from "react-icons/fa";
 
 class Book extends React.Component {
   state = {
     data: null,
     isLoading: false,
   };
 
-  getBook = async (isbn) => {
+  getBook = async (id) => {
     try {
       this.setState({ isLoading: true });
       const { data } = await axios(
-        `https://www.googleapis.com/books/v1/volumes?q=isbn:${isbn}&key=${process.env.REACT_APP_API_KEY}`
+        `https://www.googleapis.com/books/v1/volumes/${id}`
       );
-      this.setState({ data: data.items[0], isLoading: false });
+      this.setState({ data: data, isLoading: false });
+      const favorited = { ...this.state.data };
+      favorited.favorited = false;
+      this.setState({ data: favorited });
     } catch (error) {
       console.log(error);
     }
   };
 
   componentDidMount() {
-    const { isbn } = this.props.match.params;
-    this.getBook(isbn);
+    const { id } = this.props.match.params;
+    this.getBook(id);
+    let favoritedBooks =
+      JSON.parse(localStorage.getItem("favoritedBooks")) || {};
+    this.setState({ favoritedBooks });
   }
 
+  handleClick = (id) => {
+    const book = { ...this.state.data };
+    book.favorited = !book.favorited;
+    let favoritedBooks =
+      JSON.parse(localStorage.getItem("favoritedBooks")) || {};
+
+    let modifiedFavorites = [];
+    if (favoritedBooks[id]) {
+      modifiedFavorites = Object.entries(favoritedBooks).filter(
+        ([key, value]) => value.id !== id
+      );
+      const favoritesObject = Object.fromEntries(modifiedFavorites);
+      this.setState({ favoritedBooks: favoritesObject }, function () {
+        localStorage.setItem(
+          "favoritedBooks",
+          JSON.stringify(this.state.favoritedBooks)
+          );
+      });
+    } else if (!favoritedBooks) {
```

Comment:
is this condition ever going to be true?

Reply:
It doesn't appear so. It was something that you initially had set up during the call, but I think with all the other things I added, it is not needed anymore.